engine/src/mac-dialog.mm

```diff
@@ -626,12 +626,16 @@ void MCPlatformBeginFolderOrFileDialog(MCPlatformFileDialogKind p_kind, MCPlatfo
     else
     {
         [t_panel setPrompt: @"Choose"];
-        [t_panel setCanChooseFiles: NO];
-        [t_panel setCanChooseDirectories: YES];
-        [t_panel setAllowsMultipleSelection: NO];
+        [(NSOpenPanel *)t_panel setCanChooseFiles: NO];
+        [(NSOpenPanel *)t_panel setCanChooseDirectories: YES];
+        [(NSOpenPanel *)t_panel setAllowsMultipleSelection: NO];
+
+        // MM-2012-03-01: [[ BUG 10046]] Make sure the "new folder" button is enabled for folder dialogs
+        [t_panel setCanCreateDirectories: YES];
     }
     
-    if ([t_panel respondsToSelector:@selector(isAccessoryViewDisclosed)])
+    if (p_kind != kMCPlatformFileDialogKindSave && 
```

Comment:
@bwmilby not really sure what's up here unless apple decided to have NSSavePanel implement `isAccessoryViewDisclosed` just to throw an error...

Anyway... this all looks safe to rebase onto develop-8.1

Reply:
We should have both issues reported and notes for them though

Reply:
When I do the "make config-mac", there are 2 changes that I notice:
- SDK 10.8 --> 10.9
- target_arch=i386 --> i386 x86_64

I've added a new bug report and release notes.

Reply:
The accessory view problem I believe is related to SDK / Xcode combinations - I don't think it occurs in builds done from vulcan...

The folder issue is distinct and needs to be submitted in a separate PR against develop-8.1 as that is a regression against a fix made in that release.

Reply:
Should the compiler warning corrections be moved to the 8.1.7 patch too?  That would leave this PR as just adjusting the accessory view code (either as I did already or moving it into the else clause which would make it match where it was logically before the change that introduced the issue).

Reply:
I think just fix the folder bug in release-8.1.7 - then do the rest in the fix to solve the problem the PR was originally for :)